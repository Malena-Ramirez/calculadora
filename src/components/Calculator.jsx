import React from 'react';
import Display from './Display';
import NumberKeys from './NumberKeys';
import OperatorKeys from './OperatorKeys';
import { CalculatorContainer, Tittle, CalculatorMain, KeyContainer, NumberKeysContainer, OperatorKeysContainer } from './styled-components/CalculatorStyled';
import { numbers, operators } from '../data';

const Calculator = () => {
  return (
    <CalculatorContainer>
      <CalculatorMain>
        <Tittle>Calculadora</Tittle>
        <Display />
        <KeyContainer>

          <NumberKeysContainer>
            {numbers.map(number => (
              <NumberKeys id={number.id} number={number.number} key={number.id} />
            ))}
          </NumberKeysContainer>
          
          <OperatorKeysContainer>
          {
            operators.map(operator =>(
              <OperatorKeys 
                id={operator.id} 
                symbol={operator.symbol} 
                special={operator.special}
                bg={operator.bg} 
                key={operator.id} />
            ))
          }
          </OperatorKeysContainer>

        </KeyContainer>
      </CalculatorMain>
    </CalculatorContainer>
  );
};

export default Calculator;
