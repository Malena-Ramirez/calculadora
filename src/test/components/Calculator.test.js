import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from "enzyme";
import Calculator from '../../components/Calculator';

configure({ adapter: new Adapter() });
describe('Pruebas en <Calculator />', () => {

  test('Debe mostrarse correctamente', () => {
    const wrapper = shallow(<Calculator />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrarse una lista de nÃºmeros', () => {
    const numbersTest = [
      {
        id: 'uno',
        number: 1
      },
      {
        id: 'dos',
        number: 2
      },
      {
        id: 'tres',
        number: 3
      },
    ];

    const wrapper = shallow(<Calculator numbersList={numbersTest} />);
    expect(wrapper.find('NumberKeys').length).toBe(numbersTest.length);

  })

})
