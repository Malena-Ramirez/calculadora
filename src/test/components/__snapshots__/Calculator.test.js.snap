// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Calculator /> Debe mostrarse correctamente 1`] = `ShallowWrapper {}`;
